<?xml version="1.0"?>
<launch>

    <param name="use_sim_time" value="false"/>

    <node pkg="stefmap_ros" type="stefmap_node_online.py" name="stefmap_node" output="screen">
    	<param name="grid_size" value="1"/>  
		<param name="x_min" value="-10" />
    	<param name="x_max" value="60" />
    	<param name="y_min" value="-10" />
    	<param name="y_max" value="60" />
    	<param name="interval_time" value="5" />
    	<param name="num_bins" value="8" />
    	<param name="frame_id" value="world" />
    	<param name="people_detections_topic_1" value="/groundtruth/detected_persons/posearray" />
        <param name="load_data" value="False" />
        <param name="data_to_load_file" value="$(find stefmap_ros)/data/orkla_2019_06_13_histograms.txt" />
        <param name="save_histograms" value="False" />
        <param name="histograms_to_save_file" value="$(find stefmap_ros)/data/test_histograms_saved.txt" />

    </node>
    
    <node pkg="fremenarray" type="fremenarray" name="fremenarray" output="screen">
    </node>

    <node pkg="stefmap_ros" type="spencermsgs_to_posearray.py" name="spencermsgs_to_posearray" output="screen">
      <param name="people_detections_topic" value="/groundtruth/detected_persons"/>
    </node>

    <node pkg="iliad_human_perception_launch" type="groundtruth_tracks_from_gazebo.py" name="groundtruth_tracks_from_gazebo" output="screen">
    </node>


    <node pkg="stefmap_ros" type="compute_visibility_map.py" name="compute_visibility_map_node" output="screen">
        <param name="grid_size" value="1"/>  
        <param name="x_min" value="-10" />
        <param name="x_max" value="60" />
        <param name="y_min" value="-10" />
        <param name="y_max" value="60" />
        <param name="frame_id" value="world" />
        <param name="coverage_laser_topic_1" value="/robot4/sensors/laser2d_top" />
        <param name="coverage_laser_topic_2" value="/robot4/sensors/laser2d_floor" />
        <param name="coverage_time_update" value="2" />
        <param name="max_coverage_distance" value="13.5" />
        <param name="full_observability" value="False" />
    </node>
    
<!--
    <node pkg="stefmap_ros" type="stefmap_client_example.py" name="stefmap_client_node" output="screen" >
    </node>
-->

</launch>